openmc/tallies.py

```diff
@@ -1620,108 +1660,129 @@ def _align_tally_data(self, other):
+        # Repeat and tile the data by score in preparation for performing
+        # the tensor product across scores.
+        if score_product == 'tensor':
+            self._mean = np.repeat(self.mean, other.num_score_bins, axis=2)
+            self._std_dev = np.repeat(self.std_dev, other.num_score_bins, axis=2)
+            other._mean = np.tile(other.mean, (1, 1, self.num_score_bins))
+            other._std_dev = np.tile(other.std_dev, (1, 1, self.num_score_bins))
+
+        # Add scores to each tally such that each tally contains the complete set
+        # of scores necessary to perform an entrywise product. New scores added
+        # to a tally will be set to zero.
+        else:
+
+            # Get the set of scores that each tally is missing
+            other_missing_scores = set(self.scores).difference(set(other.scores))
+            self_missing_scores = set(other.scores).difference(set(self.scores))
+
+            # Add scores present in self but not in other to other
+            for score in other_missing_scores:
+                other._mean = np.insert(other.mean, other.num_score_bins, 0, axis=2)
+                other._std_dev = np.insert(other.std_dev, other.num_score_bins, 0, axis=2)
+                other.add_score(score)
+
+            # Add scores present in other but not in self to self
+            for score in self_missing_scores:
+                self._mean = np.insert(self.mean, self.num_score_bins, 0, axis=2)
+                self._std_dev = np.insert(self.std_dev, self.num_score_bins, 0, axis=2)
+                self.add_score(score)
+                self.num_score_bins += 1
+
+            # Align other scores with self scores
+            for i, score in enumerate(self.scores):
+                other_index = other.scores.index(score)
+
+                # If necessary, swap other score
+                if other_index != i:
+                    other._swap_scores(score, other.scores[i])
+
+        # Correct the stride for other filters
+        stride = other.num_nuclides * other.num_score_bins
+        for filter in reversed(other.filters):
+            filter.stride = stride
+            stride *= filter.num_bins
+
+        # Correct the stride for self filters
+        stride = self.num_nuclides * self.num_score_bins
+        for filter in reversed(self.filters):
+            filter.stride = stride
+            stride *= filter.num_bins
 
         data = {}
         data['self'] = {}
         data['other'] = {}
-        data['self']['mean'] = self_mean
-        data['other']['mean'] = other_mean
-        data['self']['std. dev.'] = self_std_dev
-        data['other']['std. dev.'] = other_std_dev
+        data['self']['mean'] = self.mean
+        data['other']['mean'] = other.mean
+        data['self']['std. dev.'] = self.std_dev
+        data['other']['std. dev.'] = other.std_dev
         return data
 
-    def swap_filters(self, filter1, filter2, inplace=False):
+    def _swap_filters(self, filter1, filter2):
```

Comment:
It doesn't look like `_swap_filters(...)` "copies this tally ..." now as is stated in the docstring. Revise the docstring to reflect the fact that the method no longer returns a copy of the tally with swapped filters, and instead does the swapping inplace.